services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/StormLcmSpec.groovy

```diff
@@ -0,0 +1,66 @@
+package org.openkilda.functionaltests.spec.resilience
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.messaging.info.event.IslChangeType
+
+import com.spotify.docker.client.DefaultDockerClient
+import com.spotify.docker.client.DockerClient
+import com.spotify.docker.client.DockerClient.ListContainersParam
+import spock.lang.Narrative
+import spock.lang.Shared
+
+@Narrative("""
+Storm Lifecycle Management: verifies system behavior during restart of certain parts of Storm. This is required to 
+simulate prod deployments, which are done on the live environment.
+""")
+class StormLcmSpec extends BaseSpecification {
+    private static final String WFM_CONTAINER_NAME = "/wfm"
+
+    @Shared
+    DockerClient dockerClient
+
+    def setupOnce() {
+        //since we simulate storm restart by restarting the docker container, for now this is only possible on virtual
+        //TODO(rtretiak): this can possibly be achieved for 'hardware' via lock-keeper instance
+        requireProfiles("virtual")
+        dockerClient = DefaultDockerClient.fromEnv().build()
+    }
+
+    /**
+     * This test takes quite some time (~7+ minutes) since it redeploys all the storm topologies.
+     * Aborting it in the middle of execution will lead to Kilda malfunction.
+     */
+    def "System survives Storm topologies restart"() {
```

Comment:
It looks like 'when' is missing.

Reply:
'restart' can be used as a noun too, in this case the sentence looks fine